### ISSUE TITLE
inconsistent term references in translated message when building python doc

### ISSUE THREAD

**Issue by Seluj78** [sphinx-doc/sphinx#8932]
**Describe the bug**
When building the python doc (translated to french), I get this error
```
/Users/seluj78/Projects/PyDocs/cpython/Doc/library/sys.rst::inconsistent term references in translated message. original: [], translated: [":ref:`événement d'audit <auditing>`"]
```

Which is weird. It doesn't say the line number anywhere and when looking through the `sys.po` file I don't find any entry that matches the error

**To Reproduce**
```shell
$ git clone https://github.com/seluj78/python-docs-fr
$ cd python-docs-fr
$ git checkout library/sys
$ python3.8 -m venv venv
$ source venv/bin/activate
$ git clone https://github.com/python/cpython ../cpython/
$ python -m pip install -r requirements.txt -r ../cpython/Doc/requirements.txt
$ pip install --upgrade sphinx==2.4.4
$ make
```

(Using sphinx 2.4.4 as our CI does)

**Expected behavior**
Give the correct line number or doesn't throw the error

**Your project**
https://github.com/python/python-docs-fr/pull/1529
https://github.com/python/python-docs-fr/pull/1529/checks?check_run_id=1971969432

**Environment info**
- OS: MacOS 10.14.6
- Python version: 3.8.1rc1
- Sphinx version: 2.4.4

pip freeze:

```
alabaster==0.7.12
Babel==2.9.0
blurb==1.0.8
certifi==2020.12.5
chardet==4.0.0
colorlog==4.7.2
docutils==0.16
gitignore-parser==0.0.8
idna==2.10
imagesize==1.2.0
Jinja2==2.11.3
MarkupSafe==1.1.1
packaging==20.9
padpo==0.10.0
pogrep==0.1.2
polib==1.1.0
pomerge==0.1.4
pospell==1.0.11
potodo==0.19.1
poutils==0.12.0
powrap==0.4.0
Pygments==2.8.0
pygrammalecte==1.3.0
pyparsing==2.4.7
python-docs-theme==2020.12
pytz==2021.1
regex==2020.11.13
requests==2.25.1
simple-term-menu==0.10.5
simplelogging==0.11.0
snowballstemmer==2.1.0
Sphinx==2.4.4
sphinxcontrib-applehelp==1.0.2
sphinxcontrib-devhelp==1.0.2
sphinxcontrib-htmlhelp==1.0.3
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.3
sphinxcontrib-serializinghtml==1.1.4
tabulate==0.8.7
tqdm==4.56.2
urllib3==1.26.3
```

**Comment by Seluj78** [comment]
@JulienPalard

**Comment by tk0miya** [comment]
On my local, I can't reproduce the error with the following Dockerfile. Is it something different with GitHub CI?
```
FROM python:3.8-slim

RUN apt update; apt install -y build-essential curl git unzip vim
RUN git clone https://github.com/seluj78/python-docs-fr
WORKDIR /python-docs-fr
RUN git clone https://github.com/python/cpython ../cpython/
RUN python -m pip install -r requirements.txt -r ../cpython/Doc/requirements.txt
RUN python -m pip install --upgrade sphinx==2.4.4
RUN make
```

**Comment by Seluj78** [comment]
You didn't checkout the correct branch!

`git checkout library/sys`

Also, here is the CI steps https://github.com/python/python-docs-fr/blob/3.9/.github/workflows/tests.yml#L62-L94

**Comment by tk0miya** [comment]
Thank you for the info. Reproduced.

And it found the reason why Sphinx does not show the line number of the paragraph. The paragraph node generated by `audit-event` directive does not have its line number. This is a fix for it.

```
diff --git a/Doc/tools/extensions/pyspecific.py b/Doc/tools/extensions/pyspecific.py
index 2fad9ec00c..74de6ffa75 100644
--- a/Doc/tools/extensions/pyspecific.py
+++ b/Doc/tools/extensions/pyspecific.py
@@ -191,6 +191,7 @@ class AuditEvent(Directive):
         info['source'].append((env.docname, target))

         pnode = nodes.paragraph(text, classes=["audit-hook"], ids=ids)
+        pnode.line = self.lineno
         if self.content:
             self.state.nested_parse(self.content, self.content_offset, pnode)
         else:
```

I still don't understand why the mismatch happened. But it would related to the directive, I guess. Please ask what happened to the author of the extension.

**Comment by JulienPalard** [comment]
@Seluj78 you can do a PR on cpython to fix it.

**Comment by Seluj78** [comment]
See https://github.com/python/cpython/pull/24692

**Comment by tk0miya** [comment]
This is already resolved via python/cpython#24692. Closing.
